homeassistant/helpers/entity_registry.py

```diff
@@ -1630,13 +1631,17 @@ def get_optional_enum[_EnumT: StrEnum](
         self.entities = entities
         self._entities_data = entities.data
 
-    @callback
     def _data_to_save(self) -> dict[str, Any]:
         """Return data of entity registry to store in a file."""
+        # Create intermediate lists to allow this method to be called from a thread
+        # other than the event loop.
         return {
-            "entities": [entry.as_storage_fragment for entry in self.entities.values()],
+            "entities": [
+                entry.as_storage_fragment for entry in list(self.entities.values())
+            ],
             "deleted_entities": [
-                entry.as_storage_fragment for entry in self.deleted_entities.values()
+                entry.as_storage_fragment
+                for entry in list(self.deleted_entities.values())
             ],
         }
```

Comment:
I think this is enough. The entity registry objects are frozen, and have no attributes which are meant to be mutated.
We should probably change mutable attributes to immutable variants, for example change the aliases to a `frozenset`: https://github.com/home-assistant/core/blob/7631c8fd45e65e699a4c880a4f8b9d849c1a81bb/homeassistant/helpers/entity_registry.py#L186